packages/graphback-codegen-schema/src/SchemaCRUDPlugin.ts

```diff
@@ -701,18 +702,27 @@ export class SchemaCRUDPlugin extends GraphbackPlugin {
     const relationOwner = relationship.ownerField.name;
     const model = modelNameToModelDefinition[modelName];
 
+    // construct a unique key to identify the dataloader
+    const dataLoaderName = `${modelName}-${relationship.kind}-${relationIdField.name}-${relationship.relationForeignKey}-${Math.random().toString(16).slice(2)}-DataLoader`;
     resolverObj[relationOwner] = (parent: any, _: any, context: GraphbackContext, info: GraphQLResolveInfo) => {
       if (!context.graphback || !context.graphback.services || !context.graphback.services[modelName]) {
         throw new Error(`Missing service for ${modelName}`);
       }
 
-      const selectedFields = getSelectedFieldsFromResolverInfo(info, model);
-      const graphback = {
-        services: context.graphback.services,
-        options: { selectedFields }
-      };
+      if (!context[dataLoaderName]) {
+        context[dataLoaderName] = new DataLoader<string, any>(async (keys: string[]) => {
+          const selectedFields = getSelectedFieldsFromResolverInfo(info, model);
```

Comment:
Love it! Thanak you for adding this. Not sure if we can test it somehow?
From user/response point if view result will be the same.

Reply:
I think once I make the integration tests pass, we should be good https://app.circleci.com/pipelines/github/aerogear/graphback/6791/workflows/c990cf6b-8a62-4513-a09f-c6961a505386/jobs/6925

I also have a TODO item about `relationships` in the benchmark folder.